src/stack_mitosis/cli.clj

```diff
@@ -50,10 +51,18 @@
       (log/infof "Assuming role %s" (:role-arn role))
       (sudo/sudo-provider role)))
   (let [rds (interpreter/client)
-        instances (interpreter/databases rds)]
-    (when (interpreter/verify-databases-exist instances [source target])
+        instances (interpreter/databases rds)
+        same-vpc (lookup/same-vpc?
+                  (lookup/by-id instances source)
+                  (lookup/by-id instances target))
```

Comment:
Oh, just noticed this ordering. I *think* the same-vpc lookup check may need to be guarded by the call to `verify-databases-exist`, as that should report an error if source or target are missing prior to comparing vpc-id. That way the user can distinguish between source or target didn't exist vs vpc-a != vpc-b because vpc-b is nil because target doesn't exist. 

I think that just means lifting the let for calculating same-vpc inside of the `when` for the `verify-databases-exist`? It's getting a little busy in this method though, I can look into splitting it up a bit afterwards to clarify. I'm happy to take a stab at it if that would help.

Reply:
Did it in the last commit. I agree it's getting busy.

We might be able to simplify in a follow-up PR by always using restore snapshot for the primaries, if you're game.

Reply:
Thanks for moving that check up, much better. I'll take a look at simplifying, appreciate that it would be less complicated if it always used snapshots, but I think would prefer if both were an option. It's possible it's always faster to use a snapshot because of changes that AWS did on the backend in the past year or two, but I know at one point replicas were much quicker than snapshots. I'm also leery about doing it two different approaches by default. Increases the chance that one or the other will stop working from disuse.

I think this feature accelerates the need for a fully automated integration test (even if it costs a few dollars to run each time). Hopefully we can get this merged in soon so other features can be added more piece-meal. Let me know if there is something here I can help with.